frontend/packages/design-system/src/components/Icon.tsx

```diff
@@ -1,163 +1,31 @@
-    case IconName.HOME:
-      iconComponent = icons.Home;
-      break;
-    case IconName.IMAGE:
-      iconComponent = icons.Image;
-      break;
-    case IconName.LOCK:
-      iconComponent = icons.Lock;
-      break;
-    case IconName.MALE:
-      iconComponent = icons.Male;
-      break;
-    case IconName.MENU:
-      iconComponent = icons.Menu;
-      break;
-    case IconName.MORE:
-      iconComponent = icons.More;
-      break;
-    case IconName.NEXT:
-      iconComponent = icons.Next;
-      break;
-    case IconName.NOTIFICATION:
-      iconComponent = icons.Notification;
-      break;
-    case IconName.PLUS:
-      iconComponent = icons.Plus;
-      break;
-    case IconName.PRINT:
-      iconComponent = icons.Print;
-      break;
-    case IconName.PROTECTION:
-      iconComponent = icons.Protection;
-      break;
-    case IconName.REGISTRATION:
-      iconComponent = icons.Registration;
-      break;
-    case IconName.RELOAD:
-      iconComponent = icons.Reload;
-      break;
-    case IconName.SAVE:
-      iconComponent = icons.Save;
-      break;
-    case IconName.SEARCH:
-      iconComponent = icons.Search;
-      break;
-    case IconName.SETTING:
-      iconComponent = icons.Setting;
-      break;
-    case IconName.SHARE:
-      iconComponent = icons.Share;
-      break;
-    case IconName.SYNC:
-      iconComponent = icons.Sync;
-      break;
-    case IconName.THUMBNAIL:
-      iconComponent = icons.Thumbnail;
-      break;
-    case IconName.UNLOCK:
-      iconComponent = icons.Unlock;
-      break;
-    case IconName.UPLOAD:
-      iconComponent = icons.Upload;
-      break;
-    case IconName.USER_ADD:
-      iconComponent = icons.UserAdd;
-      break;
-    case IconName.USER_DELETE:
-      iconComponent = icons.UserDelete;
-      break;
-    case IconName.USER_GROUP:
-      iconComponent = icons.UserGroup;
-      break;
-    case IconName.USER_GROUP_ADD:
-      iconComponent = icons.UserGroupAdd;
-      break;
-    case IconName.WHATSAPP:
-      iconComponent = icons.Whatsapp;
-      break;
-    case IconName.ACTIVITY_CAMP:
-      iconComponent = icons.ActivityCamp;
-      break;
-    case IconName.ACTIVITY_EDUCATION:
-      iconComponent = icons.ActivityEducation;
-      break;
-    case IconName.ACTIVITY_FOOD:
-      iconComponent = icons.ActivityFood;
-      break;
-    case IconName.ACTIVITY_ICLA:
-      iconComponent = icons.ActivityIcla;
-      break;
-    case IconName.ACTIVITY_SHELTER:
-      iconComponent = icons.ActivityShelter;
-      break;
-    case IconName.ACTIVITY_WASH:
-      iconComponent = icons.ActivityWash;
-      break;
-    default:
-      iconComponent = () => {
-        return <Path />;
-      };
-  }
+const Icon: FC<Props> = (props) => {
+  const { name } = props;
+  const MappedIcon = iconMap[name];
 
   return (
-    <IconNB viewBox="0 0 40 40" style={{ height: '40px', width: '40px', display: 'flex' }} children={iconComponent(variant)} />
+    <IconNB
+      viewBox="0 0 40 40"
```

Comment:
I don't think we want to be setting any props here. These should all be set through native base.

Reply:
but we do want a fix size, don't we?

Reply:
I don't think so. In my experience icon sizes vary. If it is this way in the designs now I don't think we should restrict ourselves to it. If we think there's a fixed size we want I would set the `defaultProps` for `Icon` in the theme.

Reply:
that's how I understood the designers, but okay, then I'll set some defaults